feat: do not log secret variables

vmaerten opened the pull request:
# 🔐 Add secret variable masking support

## Problem
Sensitive values (API keys, passwords, tokens) were exposed in Task command logs, creating security risks when logs are shared or stored.
I’ve done a few talks on Task, and people always ask about secret variables. This fixes https://github.com/go-task/task/issues/2066 

## Solution
This PR adds a secret: true flag for variables that masks their values as ***** in command logs while keeping the actual execution unchanged.

## Key features:
  - 🎭 Template-based masking prevents false positives (e.g., SECRET="echo" won't mask the echo command)
  - 🔄 Works with all variable types: value, sh, ref, and task-level vars
  - ⏱ Supports both regular and deferred commands
  - 🚀 Zero performance impact on fingerprinting/caching

## Example
```yml
  Taskfile.yml:
  vars:
    API_KEY:
      value: "secret-key-123"
      secret: true
    APP_NAME: "myapp"

  tasks:
    deploy:
      cmds:
        - echo "Deploying {{.APP_NAME}} with key {{.API_KEY}}"
```
  Before:
  task: [deploy] echo "Deploying myapp with key secret-key-123"

  After:
  task: [deploy] echo "Deploying myapp with key *****"

## Changes

  - Added Secret field to ast.Var struct with YAML parsing support
  - Created templater.MaskSecrets() using Go template engine for robust masking
  - Added CmdTemplate field to preserve original templates for masking
  - Comprehensive test coverage including deferred commands

## Documentation

  - Updated user guide with examples and security warnings
  - Updated schema reference

vmaerten committed:
feat: do not log secret variables

vmaerten committed:
update schema

vmaerten committed:
format

trulede commented:
Internally, a secret might be the same base type as a vars & envar (both of which are the same type internally) . So the implementation would also be clean, and none of the capability's you mention are lost (they are essential). But then you do have a bit more capability programmatically, in dealing with those objects (reflection etc). I think you are correct in those observations.

Details aside, the GitHub way is more familiar, from a user perspective.

vmaerten committed:
add env example

vmaerten committed:
format

vmaerten committed:
refactor

vmaerten committed:
refactor: compute masked command at compile time

Move secret masking from runtime (task.go) to compile time (variables.go).
This avoids recalculating variables on each log.

- Add MaskSecretsWithExtra for loop vars and deferred commands
- Rename CmdTemplate to LogCmd (clearer intent)
- Simplify logging in runCommand

vmaerten committed:
fix: align struct fields

vmaerten committed:
Trigger Build

vmaerten committed:
Merge branch 'main' into task-secret

vmaerten committed:
fix: prevent secret variable leaks in summary, verbose and key ordering

- mask secret values in `task --summary` (commands and vars listing)
- mask resolved value of dynamic (sh) secrets in verbose logs
- use masked command for platform-skipped verbose log
- allow `secret` key in any position in a var definition (not only first)
- add `value` to the JSON schema var definition
- skip masking pass when no secret is present and dedup mask helpers
- document that the `secret` flag is not propagated to derived variables
